tests/components/autozoomControl.spec.ts

```diff
@@ -524,4 +552,45 @@ describe('AutozoomControl', () => {
       });
     });
   });
+  describe('#setMode', () => {
+    let autoozoomStatusStub;
+    let sendReceiveStub;
+    beforeEach(() => {
+      autoozoomStatusStub = sinon
+        .stub(autozoomControl._deviceManager.api, 'getAutozoomStatus')
+        .resolves({ 'autozoom-mode': AutozoomModes.NORMAL });
+      sendReceiveStub = sinon.stub(deviceManager.api, 'sendAndReceiveMessagePack');
+    });
+    afterEach(() => {
+      autoozoomStatusStub.restore();
+      sendReceiveStub.restore();
+    });
+    describe('attempting to set already running mode', () => {
+      it('should log info and return without changing mode', async () => {
+        const returnMode = await autozoomControl.setMode(AutozoomModes.NORMAL);
+        expect(infoStub).to.have.been.calledOnce;
+        expect(returnMode).to.equal(AutozoomModes.NORMAL);
+        expect(sendReceiveStub).to.have.callCount(0);
+      });
+    });
+    describe('setting new mode on camera', () => {
+      it('should throw error if response from camera is not the new mode', async () => {
+        const newMode = AutozoomModes.PLAZA;
+        sendReceiveStub.resolves({ 'autozoom-mode': AutozoomModes.NORMAL });
+        let error;
+        try {
+          await autozoomControl.setMode(newMode);
+        } catch (err) {
+          error = err;
+        }
+        expect(error).instanceof(Error);
+      });
+      it('should set and return new mode if successful', async () => {
+        const newMode = AutozoomModes.PLAZA;
+        sendReceiveStub.resolves({'autozoom-mode': AutozoomModes.PLAZA});
+        const response = await autozoomControl.setMode(newMode);
+        expect(response).to.equal(newMode);
```

Comment:
Consider adding case for unsupported mode (even though it's not necessary  right now if the caller uses typescript)